mock_test.go

```diff
@@ -319,3 +319,67 @@ func TestTickerFunc_LongCallback(t *testing.T) {
 	}
 	w.MustWait(testCtx)
 }
+
+func Test_MultipleTraps(t *testing.T) {
+	t.Parallel()
+	testCtx, testCancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer testCancel()
+	mClock := quartz.NewMock(t)
+
+	trap0 := mClock.Trap().Now("0")
+	defer trap0.Close()
+	trap1 := mClock.Trap().Now("1")
+	defer trap1.Close()
+
+	timeCh := make(chan time.Time)
+	go func() {
+		timeCh <- mClock.Now("0", "1")
+	}()
+
+	c0 := trap0.MustWait(testCtx)
+	mClock.Advance(time.Second)
+	// the two trapped call instances need to be released on separate goroutines since they each wait for the Now() call
+	// to return, which is blocked on both releases happening. If you release them on the same goroutine, in either
+	// order, it will deadlock.
```

Comment:
This is kind of unfortunate. It'd be nice if `Release` also took a context, considering it can be blocking.

I wonder if there's a case to be made for erroring or warning if multiple traps can capture the same call? Perhaps quartz could also detect deadlocks and have a built-in timeout that fails the test.

Reply:
I do like the idea of `Release()` taking a context, and adding a `MustRelease(ctx)` which fails the test. Would be API breaking change, tho, unless we only did the latter. Personally I'd rather just break the API to make something safer. It's a real easy upgrade.

In the PR above this in the stack I add some logging, so you get to see that multiple traps capture, and we can make the error messages if your context expires suggest that people check for multiple captures.